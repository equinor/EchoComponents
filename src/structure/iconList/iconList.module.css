.wrapper {
    width: 100%;
    display: flex;
    flex-direction: column;
}

.header {
    width: 100%;
    background-color: var(--searchBarBackground);
    height: 48px;
    border-bottom: 2px solid var(--default);
    display: flex;
    align-items: center;
}

.header p {
    margin-left: var(--medium) !important;
}
