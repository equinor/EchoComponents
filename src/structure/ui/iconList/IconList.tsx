import { themeConst } from '@equinor/echo-framework';
import { Typography } from '@equinor/eds-core-react';
import cx from 'classnames';
import React, { CSSProperties, useCallback, useEffect, useState } from 'react';
import {
    DraggableHandleSelector,
    DraggableOrder,
    DragItem
} from '../../../elements/functional/draggableOrder/DraggableOrder';
import { Icon } from '../../../elements/icons/icon/Icon';
import styles from './iconList.module.css';

export interface IconListProps {
    style: CSSProperties;
    title: string;
    items: IconListItem[];
    isMovable: boolean;
    expanded: IconItem[];
}

export interface IconListItem {
    title: string;
    subTitle: string;
    icons: IconItem[];
}

export interface IconItem {
    icon: string;
    onClick: () => void;
}

export const IconList: React.FC<IconListProps> = ({
    style,
    title,
    items,
    isMovable,
    expanded
}: IconListProps): JSX.Element => {
    const [orderedItems, setOrderedItems] = useState<IconListItem[]>(items);
    const [expandedIndex, setExpandedIndex] = useState<number>();
    const [rowElements, setRowElements] = useState<JSX.Element[]>([]);
    useEffect(() => {
        console.log('Props items', items);
        setOrderedItems(items);
    }, [items]);

    function reorderOnChange(oldIndex: number, newIndex: number): void {
        const updatedOrder = [];

        if (newIndex > oldIndex) {
            for (let i = 0; i < oldIndex; i++) {
                updatedOrder[i] = orderedItems[i];
            }
            for (let i = oldIndex; i < newIndex; i++) {
                updatedOrder[i] = orderedItems[i + 1];
            }
            updatedOrder[newIndex] = orderedItems[oldIndex];
            for (let i = newIndex + 1; i < orderedItems.length; i++) {
                updatedOrder[i] = orderedItems[i];
            }
        } else if (newIndex < oldIndex) {
            for (let i = orderedItems.length - 1; i > oldIndex; i--) {
                updatedOrder[i] = orderedItems[i];
            }
            for (let i = oldIndex; i > newIndex; i--) {
                updatedOrder[i] = orderedItems[i - 1];
            }
            updatedOrder[newIndex] = orderedItems[oldIndex];
            for (let i = newIndex - 1; i >= 0; i--) {
                updatedOrder[i] = orderedItems[i];
            }
        }

        console.log('Reordered', updatedOrder, oldIndex, newIndex);
        updateExpandedIndex();
        setOrderedItems(updatedOrder);

        function updateExpandedIndex(): void {
            if (typeof expandedIndex === 'undefined') return;
            console.log('exp', oldIndex, newIndex, expandedIndex);
            if (oldIndex === expandedIndex) {
                setExpandedIndex(newIndex);
            } else if (newIndex > oldIndex) {
                if (oldIndex < expandedIndex && newIndex >= expandedIndex) {
                    setExpandedIndex(expandedIndex - 1);
                }
            } else if (newIndex < oldIndex) {
                if (oldIndex > expandedIndex && newIndex <= expandedIndex) {
                    setExpandedIndex(expandedIndex + 1);
                }
            }
        }
    }

    const createRows = useCallback(() => {
        const rows: JSX.Element[] = [];
        for (let i = 0; i < orderedItems.length; i++) {
            rows.push(createRow(i));
        }
        setRowElements(rows);
        console.log('Redraw', rows);

        function createRow(index: number): JSX.Element {
            return (
                <div className={styles.item} key={index}>
                    <div className={styles.row1}>
                        <div className={styles.col1}>
                            {isMovable && (
                                <div className={cx(styles.moveable, DraggableHandleSelector)}>
                                    <Icon name="unfold_more" title="Move" color={themeConst.asBuilt}></Icon>
                                </div>
                            )}
                            <div className={cx(styles.info, isMovable ? '' : styles.infoMargin)}>
                                <Typography variant="body_short" bold>
                                    {orderedItems[index].title}
                                </Typography>
                                <Typography variant="body_short">{orderedItems[index].subTitle}</Typography>
                            </div>
                        </div>
                        <div className={styles.col2}>
                            {orderedItems[index].icons.map((icon, index2) => {
                                return (
                                    <div
                                        className={styles.iconWrapper}
                                        key={index2}
                                        onClick={(): void => {
                                            icon.onClick();
                                        }}
                                    >
                                        <Icon name={icon.icon} title={icon.icon} color={themeConst.asBuilt}></Icon>
                                    </div>
                                );
                            })}
                            {expanded && (
                                <div
                                    className={styles.expandable}
                                    onClick={(): void => {
                                        console.log('Expand', index);
                                        setExpandedIndex(index);
                                    }}
                                >
                                    <Icon name="more_vertical" title="Expand" color={themeConst.asBuilt}></Icon>
                                </div>
                            )}
                        </div>
                    </div>
                    {expandedIndex == index && (
                        <div className={styles.row2}>
                            {expanded.map((icon, index3) => {
                                return (
                                    <div
                                        className={styles.iconWrapper}
                                        key={index3}
                                        onClick={(): void => {
                                            icon.onClick();
                                        }}
                                    >
                                        <Icon name={icon.icon} title={icon.icon} color={themeConst.asBuilt}></Icon>
                                    </div>
                                );
                            })}
                            <div
                                className={styles.close}
                                onClick={(): void => {
                                    setExpandedIndex(undefined);
                                }}
                            >
                                <Icon name={'close'} title={'Close'} color={themeConst.asBuilt}></Icon>
                            </div>
                        </div>
                    )}
                </div>
            );
        }
    }, [orderedItems, isMovable, expanded, expandedIndex]);

    useEffect(() => {
        createRows();
    }, [items, createRows]);

    return (
        <div className={styles.iconList} style={style}>
            <div className={styles.header}>
                <Typography group="table" variant="cell_text" bold>
                    {title}
                </Typography>
            </div>
            <DraggableOrder
                elements={rowElements}
                style={style}
                onChange={(
                    newDragItems: DragItem[],
                    newElements: JSX.Element[],
                    oldIndex: number | undefined,
                    newIndex: number | undefined
                ): void => {
                    setRowElements(newElements);
                    if (typeof oldIndex !== 'undefined' && typeof newIndex !== 'undefined' && newIndex !== oldIndex) {
                        reorderOnChange(oldIndex, newIndex);
                    }
                }}
            ></DraggableOrder>
        </div>
    );
};
