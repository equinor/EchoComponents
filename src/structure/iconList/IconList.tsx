import { Typography } from '@equinor/eds-core-react';
import React, { CSSProperties, useCallback, useEffect, useState } from 'react';
import { DraggableItem, IconItem, IconListItem } from '../..';
import { DraggableItemsWrapper } from '../../elements/draggableItemsWrapper/DraggableItemsWrapper';
import { createListRow } from './createListRow';
import styles from './iconList.module.css';

export interface IconListProps {
    style?: CSSProperties;
    title: string;
    items: IconListItem[];
    isDraggable: boolean;
    expandedIcons?: IconItem[];
    onUpdatedOrder: (itemsInNewOrder: IconListItem[]) => void;
}

export interface IconListState {
    orderedItems: IconListItem[];
    expandedIndex: number;
}
/**
 * Component that renders an IconList of rows, that can be draggable and/or expandable
 *
 * @param {IconListProps} {
 *     style: the style to override the IconList wrapper
 *     title: The title of the list
 *     items: { The items to be rendered in the row
 *          title: the title of the item in the row
 *          subtitle: the subtitle of the item in the row
 *          icons: the icons that will be displayed on the same row, and their respective actions
 *     }[]
 *     isDraggable: flag to indicate if the elements in the list should be draggable
 *     expandedIcons: list of icons that should be displayed if a row is expanded. If undefined no row can be expanded
 *     onUpdatedOrder: method that will be called when the order of the list has changed
 * }
 * @return {*}  {JSX.Element} a list of items that can be draggable and/or expandable
 */
export const IconList: React.FC<IconListProps> = ({
    style,
    title,
    items,
    isDraggable,
    expandedIcons,
    onUpdatedOrder
}: IconListProps): JSX.Element => {
    const [iconListState, setIconListState] = useState<IconListState>({ orderedItems: items, expandedIndex: -1 });

    const [rowElements, setRowElements] = useState<React.ReactNode[]>(
        iconListState.orderedItems.map((_, index) =>
            createListRow(isDraggable, index, iconListState, setIconListState, expandedIcons)
        )
    );

    useEffect(() => {
        setIconListState({ orderedItems: items, expandedIndex: -1 });
    }, [items]);

    const reorderOnChange = (newDragItems: DraggableItem[]): void => {
        const updatedOrder = newDragItems.map((e) => {
            return iconListState.orderedItems[e.id];
        });

        let expandedIndex = iconListState.expandedIndex;
        if (iconListState.expandedIndex !== -1) {
            expandedIndex = updatedOrder.findIndex((e) => e === iconListState.orderedItems[expandedIndex]);
        }

        setIconListState({ orderedItems: updatedOrder, expandedIndex });
        onUpdatedOrder(updatedOrder);
    };

    const createRows = useCallback(() => {
        setRowElements(
            iconListState.orderedItems.map((_, index) =>
                createListRow(isDraggable, index, iconListState, setIconListState, expandedIcons)
            )
        );
    }, [iconListState, isDraggable, expandedIcons]);

    useEffect(() => {
        createRows();
    }, [items, createRows]);

    return (
        <div className={styles.wrapper} style={style}>
            <div className={styles.header}>
                <Typography group="table" variant="cell_text" bold>
                    {title}
                </Typography>
            </div>
            {isDraggable ? (
                <DraggableItemsWrapper
                    style={style}
                    onChange={(newDragItems: DraggableItem[]): void => {
                        setRowElements(newDragItems.map((d) => d.element));
                        reorderOnChange(newDragItems);
                    }}
                >
                    {rowElements}
                </DraggableItemsWrapper>
            ) : (
                <div style={style}>{rowElements}</div>
            )}
        </div>
    );
};
