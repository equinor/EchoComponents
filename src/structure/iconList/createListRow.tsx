import React from 'react';
import { IconItem } from '../..';
import { ListRow } from '../../components/listRow/ListRow';
import { IconListState } from './IconList';
/**
 * Helper method that will return a list row given some input values. Used in IconList component for rendering rows
 *
 * @param {boolean} isDraggable flag to indicate that the row is draggable
 * @param {number} index the index of the list
 * @param {IconListState} iconListState the state from the IconList component used to create the row and its interactions
 * @param {(value: React.SetStateAction<IconListState>) => void} setIconListState the action to update the state from the IconList component
 * @param {IconItem[]} expandedIcons the icons to be displayed if the row is expandable. Is undefined if the row should not be expandable
 * @return {*}  {React.ReactNode} A list row component created from the input parameters
 */
export const createListRow = (
    isDraggable: boolean,
    index: number,
    iconListState: IconListState,
    setIconListState: (value: React.SetStateAction<IconListState>) => void,
    expandedIcons?: IconItem[]
): React.ReactNode => {
    return (
        <ListRow
            isDraggable={isDraggable}
            item={iconListState.orderedItems[index]}
            expandable={
                expandedIcons
                    ? {
                          expanded: iconListState.expandedIndex == index,
                          iconItems: expandedIcons,
                          setExpanded: (status: boolean): void => {
                              if (status) {
                                  setIconListState({ orderedItems: iconListState.orderedItems, expandedIndex: index });
                              } else {
                                  setIconListState({ orderedItems: iconListState.orderedItems, expandedIndex: -1 });
                              }
                          }
                      }
                    : undefined
            }
            rowIndex={index}
            key={index}
        />
    );
};
