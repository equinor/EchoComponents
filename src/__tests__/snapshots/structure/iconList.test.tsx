import React from 'react';
import renderer from 'react-test-renderer';
import { IconListItem } from '../../..';
import { IconList } from '../../../structure/iconList/IconList';

it('renders IconList correctly', () => {
    const items: IconListItem[] = [
        {
            title: 'Item 0',
            icons: [
                {
                    icon: 'link',
                    onClick: jest.fn()
                },
                {
                    icon: 'delete_to_trash',
                    onClick: jest.fn()
                }
            ]
        },
        {
            subTitle: 'Subtitle 1',
            icons: [
                {
                    icon: 'link',
                    onClick: jest.fn()
                },
                {
                    icon: 'delete_to_trash',
                    onClick: jest.fn()
                }
            ]
        },
        {
            icons: [
                {
                    icon: 'link',
                    onClick: jest.fn()
                },
                {
                    icon: 'delete_to_trash',
                    onClick: jest.fn()
                }
            ]
        },
        {
            title: 'Item 3',
            subTitle: 'Subtitle 3',
            icons: []
        },
        {
            title: 'Item 4',
            subTitle: 'Subtitle 4',
            icons: [
                {
                    icon: 'link',
                    onClick: jest.fn()
                },
                {
                    icon: 'delete_to_trash',
                    onClick: jest.fn()
                }
            ]
        }
    ];
    const tree = renderer
        .create(
            <IconList
                title={'This is the title'}
                items={items}
                isDraggable={true}
                expandedIcons={['link', 'delete_to_trash'].map((i) => {
                    return {
                        icon: i,
                        onClick: jest.fn()
                    };
                })}
                onUpdatedOrder={jest.fn()}
            ></IconList>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
