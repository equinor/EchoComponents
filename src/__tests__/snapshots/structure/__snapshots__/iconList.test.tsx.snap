// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders IconList correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="header"
  >
    <p
      className="sc-hKgILt gmaETi"
    >
      This is the title
    </p>
  </div>
  <div>
    <div>
      <div
        className="globalDraggableHandle"
        data-id="0"
      >
        <div
          className="wrapper"
        >
          <div
            className="row"
          >
            <div
              className="infoCol"
            >
              <div
                className="draggable"
              >
                <svg
                  aria-labelledby="eds-unfold_more-1"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Move"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-unfold_more-1"
                  >
                    Move
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M15.17 9L12 5.83 8.83 9 7.41 7.59 12 3l4.58 4.59L15.17 9zm-6.34 6L12 18.17 15.17 15l1.42 1.41L12 21l-4.58-4.59L8.83 15z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </div>
              <div
                className="info"
              >
                <p
                  className="sc-hKgILt bCyDSy"
                >
                  Item 0
                </p>
              </div>
            </div>
            <div
              className="iconCol"
            >
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-link-2"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="link"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-link-2"
                  >
                    link
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M11 15H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2zm6-8h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5zm-1 4H8v2h8v-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-delete_to_trash-3"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="delete_to_trash"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-delete_to_trash-3"
                  >
                    delete_to_trash
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M14.5 3h-5l-1 1H5v2h14V4h-3.5l-1-1zM16 9v10H8V9h8zM6 7h12v12c0 1.1-.9 2-2 2H8c-1.1 0-2-.9-2-2V7z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[Function]}
              >
                <svg
                  aria-labelledby="eds-more_vertical-4"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Expand"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-more_vertical-4"
                  >
                    Expand
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-2 8c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="globalDraggableHandle"
        data-id="1"
      >
        <div
          className="wrapper"
        >
          <div
            className="row"
          >
            <div
              className="infoCol"
            >
              <div
                className="draggable"
              >
                <svg
                  aria-labelledby="eds-unfold_more-5"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Move"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-unfold_more-5"
                  >
                    Move
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M15.17 9L12 5.83 8.83 9 7.41 7.59 12 3l4.58 4.59L15.17 9zm-6.34 6L12 18.17 15.17 15l1.42 1.41L12 21l-4.58-4.59L8.83 15z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </div>
              <div
                className="info"
              >
                <p
                  className="sc-hKgILt dMHJLh"
                >
                  Subtitle 1
                </p>
              </div>
            </div>
            <div
              className="iconCol"
            >
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-link-6"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="link"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-link-6"
                  >
                    link
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M11 15H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2zm6-8h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5zm-1 4H8v2h8v-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-delete_to_trash-7"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="delete_to_trash"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-delete_to_trash-7"
                  >
                    delete_to_trash
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M14.5 3h-5l-1 1H5v2h14V4h-3.5l-1-1zM16 9v10H8V9h8zM6 7h12v12c0 1.1-.9 2-2 2H8c-1.1 0-2-.9-2-2V7z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[Function]}
              >
                <svg
                  aria-labelledby="eds-more_vertical-8"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Expand"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-more_vertical-8"
                  >
                    Expand
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-2 8c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="globalDraggableHandle"
        data-id="2"
      >
        <div
          className="wrapper"
        >
          <div
            className="row"
          >
            <div
              className="infoCol"
            >
              <div
                className="draggable"
              >
                <svg
                  aria-labelledby="eds-unfold_more-9"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Move"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-unfold_more-9"
                  >
                    Move
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M15.17 9L12 5.83 8.83 9 7.41 7.59 12 3l4.58 4.59L15.17 9zm-6.34 6L12 18.17 15.17 15l1.42 1.41L12 21l-4.58-4.59L8.83 15z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </div>
              <div
                className="info"
              />
            </div>
            <div
              className="iconCol"
            >
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-link-10"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="link"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-link-10"
                  >
                    link
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M11 15H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2zm6-8h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5zm-1 4H8v2h8v-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-delete_to_trash-11"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="delete_to_trash"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-delete_to_trash-11"
                  >
                    delete_to_trash
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M14.5 3h-5l-1 1H5v2h14V4h-3.5l-1-1zM16 9v10H8V9h8zM6 7h12v12c0 1.1-.9 2-2 2H8c-1.1 0-2-.9-2-2V7z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[Function]}
              >
                <svg
                  aria-labelledby="eds-more_vertical-12"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Expand"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-more_vertical-12"
                  >
                    Expand
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-2 8c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="globalDraggableHandle"
        data-id="3"
      >
        <div
          className="wrapper"
        >
          <div
            className="row"
          >
            <div
              className="infoCol"
            >
              <div
                className="draggable"
              >
                <svg
                  aria-labelledby="eds-unfold_more-13"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Move"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-unfold_more-13"
                  >
                    Move
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M15.17 9L12 5.83 8.83 9 7.41 7.59 12 3l4.58 4.59L15.17 9zm-6.34 6L12 18.17 15.17 15l1.42 1.41L12 21l-4.58-4.59L8.83 15z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </div>
              <div
                className="info"
              >
                <p
                  className="sc-hKgILt bCyDSy"
                >
                  Item 3
                </p>
                <p
                  className="sc-hKgILt dMHJLh"
                >
                  Subtitle 3
                </p>
              </div>
            </div>
            <div
              className="iconCol"
            >
              <button
                className="iconButton"
                onClick={[Function]}
              >
                <svg
                  aria-labelledby="eds-more_vertical-14"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Expand"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-more_vertical-14"
                  >
                    Expand
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-2 8c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="globalDraggableHandle"
        data-id="4"
      >
        <div
          className="wrapper"
        >
          <div
            className="row"
          >
            <div
              className="infoCol"
            >
              <div
                className="draggable"
              >
                <svg
                  aria-labelledby="eds-unfold_more-15"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Move"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-unfold_more-15"
                  >
                    Move
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M15.17 9L12 5.83 8.83 9 7.41 7.59 12 3l4.58 4.59L15.17 9zm-6.34 6L12 18.17 15.17 15l1.42 1.41L12 21l-4.58-4.59L8.83 15z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </div>
              <div
                className="info"
              >
                <p
                  className="sc-hKgILt bCyDSy"
                >
                  Item 4
                </p>
                <p
                  className="sc-hKgILt dMHJLh"
                >
                  Subtitle 4
                </p>
              </div>
            </div>
            <div
              className="iconCol"
            >
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-link-16"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="link"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-link-16"
                  >
                    link
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M11 15H7c-1.65 0-3-1.35-3-3s1.35-3 3-3h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-2zm6-8h-4v2h4c1.65 0 3 1.35 3 3s-1.35 3-3 3h-4v2h4c2.76 0 5-2.24 5-5s-2.24-5-5-5zm-1 4H8v2h8v-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[MockFunction]}
              >
                <svg
                  aria-labelledby="eds-delete_to_trash-17"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="delete_to_trash"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-delete_to_trash-17"
                  >
                    delete_to_trash
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M14.5 3h-5l-1 1H5v2h14V4h-3.5l-1-1zM16 9v10H8V9h8zM6 7h12v12c0 1.1-.9 2-2 2H8c-1.1 0-2-.9-2-2V7z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
              <button
                className="iconButton"
                onClick={[Function]}
              >
                <svg
                  aria-labelledby="eds-more_vertical-18"
                  className="sc-kfzAmx blqdWU"
                  fill="#007079"
                  height="24px"
                  name={null}
                  role="img"
                  title="Expand"
                  viewBox="0 0 24 24"
                  width="24px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title
                    id="eds-more_vertical-18"
                  >
                    Expand
                  </title>
                  <path
                    className="sc-fKFyDc"
                    clipRule="evenodd"
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-2 8c0-1.1.9-2 2-2s2 .9 2 2-.9 2-2 2-2-.9-2-2z"
                    data-testid="eds-icon-path"
                    fillRule="evenodd"
                    height="24"
                    size={null}
                    transform={null}
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
